rn/providers/TasksProvider.js

```diff
@@ -0,0 +1,113 @@
+import React, { useContext, useState, useEffect, useRef } from "react";
+import Realm from "realm";
+import { Task } from "../schemas";
+import { useAuth } from "./AuthProvider";
+
+const TasksContext = React.createContext(null);
+
+const TasksProvider = ({ children, projectPartition }) => {
+  const [tasks, setTasks] = useState([]);
+  const { user } = useAuth();
+  const realmRef = useRef(null);
+
+  const createTask = (newTaskName) => {
+    const projectRealm = realmRef.current;
+    projectRealm.write(() => {
+      // Create a new task in the same partition -- that is, in the same project.
+      projectRealm.create(
+        "Task",
+        new Task({
+          name: newTaskName || "New Task",
+          partition: projectPartition,
+        })
+      );
+    });
+  };
+
+  const setTaskStatus = (task, status) => {
+    // One advantage of centralizing the realm functionality in this provider is
+    // that we can check to make sure a valid status was passed in here.
+    if (
+      ![
+        Task.STATUS_OPEN,
+        Task.STATUS_IN_PROGRESS,
+        Task.STATUS_COMPLETE,
+      ].includes(status)
+    ) {
+      throw new Error(`Invalid Status ${status}`);
+    }
+    const projectRealm = realmRef.current;
+    projectRealm.write(() => {
+      task.status = status;
+    });
+  };
+
+  // Define the function for deleting a task.
+  const deleteTask = (task) => {
+    const projectRealm = realmRef.current;
+    projectRealm.write(() => {
+      projectRealm.delete(task);
+      setTasks([...projectRealm.objects("Task").sorted("name")]);
+    });
+  };
+
+  useEffect(() => {
+    const openRealm = async () => {
+      const config = {
+        sync: {
+          user: user,
+          partitionValue: projectPartition,
+        },
+      };
+      // open a realm for this particular project
+      const projectRealm = await Realm.open(config);
+      realmRef.current = projectRealm;
+
+      const syncTasks = projectRealm.objects("Task");
+      let sortedTasks = syncTasks.sorted("name");
+      setTasks([...sortedTasks]);
+      sortedTasks.addListener(() => {
+        setTasks([...sortedTasks]);
+      });
+    };
+    openRealm();
+
+    // If there is an open realm, we must close it.
+    const projectRealm = realmRef.current;
```

Comment:
Is it actually possible for there to be an open realm here? The dependency list of the effect is empty, so it will never be triggered twice.

That said, I don't think that's the intended behavior. The dependency list should probably include `projectPartition`. I think it just happens to work because the navigation controller seems to be fully unmounting and remounting the TasksProvider every time.

Reply:
Fixed - the dependency list includes ```projectPartition``` now.